compat/src/index.d.ts

```diff
@@ -748,7 +748,7 @@ declare namespace React {
 	): void;
 
 	export type PropsWithChildren<P = unknown> = P & {
-		children?: preact.ComponentChild | undefined;
```

Comment:
[React's type](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/69db0535918d81978243eda2425b41c6afe81a63/types/react/index.d.ts#L1624) seems to be `ReactNode`, rather than `ReactNode[]` (which they do use elsewhere). As `ComponentChildren` is `ComponentChild | ComponentChild[]`, I'm not quite sure this change is correct.

Does this work correctly in React? It's very possible we have an issue elsewhere in the types.

Reply:
well, yes and no: `ReactNode` is an alias of `ComponentChild` but does not include `Iterable<ComponentChild>` like `ReactNode` does and simply extending that union in `preact/compat` only isn't possible the way it's written currently. I thought this is the minimal change but I agree, imho the correct change would be to drop `ComponentChildren`, rename `ComponentChild` to `PreactNode` and include `Iterable<PreactNode>` in it. But that's not exactly a small change...

Reply:
We don't support iterables, the type is correct there FWIW.

Reply:
wdym "we don't support iterables"? You have arrays in your types, arrays are iterables? Or did you mean `PreactNode[]` instead of `PreactNode`?

Reply:
as for "does this work correctly in react" - well, it's a type change? Ofc react still works. And if you're using react (not preact) then this issue wouldn't arise in the first place?

Kinda confused what you mean to say here... changing this solves the described issue and doesn't break anything else in the rather large project that I'm currently migrating from react to preact, so... yes? It works in a "react like context"?

Reply:
> wdym "we don't support iterables"? You have arrays in your types, arrays are iterables?

Arrays are iterable, but not all iterables are arrays; `Map`, `Set`, `Uint32Array`, and other structures are iterable too.

> as for "does this work correctly in react" - well, it's a type change? Ofc react still works.

I'm asking if your type (for your component) works in React -- I've always used `ReactNode[]`/`ComponentChildren` for multiple nodes, personally.

Reply:
Yes it does, we're using this heavily throughout multiple large projects and never had any problems until we migrated to preact (which we did because react is basically incompatible with CEs and probably will continue to be)